model/dependency.go

```diff
@@ -91,6 +91,30 @@ func (di *dependencyIncluder) handle(pair TVPair, activationInfo *specificActiva
 		return true, nil
 	}
 
+	// For a task group task inside a single host task group, the previous
+	// tasks in the task group are implicit dependencies of the current task.
+	if tg := di.Project.FindTaskGroupForTask(pair.Variant, pair.TaskName); tg != nil && tg.MaxHosts == 1 {
+		catcher := grip.NewBasicCatcher()
+		for _, t := range tg.Tasks {
+			// When we reach the current task, stop recursing.
+			if t == pair.TaskName {
+				break
+			}
+			_, err := di.handle(TVPair{pair.Variant, t}, activationInfo, generatedVariants, false)
```

Comment:
nit: it's generally preferable to specify the fields in the struct rather than rely on the order of field declaration in the `TVPair` definition. If the fields are ever changed switching the field order, this can break.